src/pu-emmc.c

```diff
@@ -881,5 +896,55 @@ pu_emmc_new(const gchar *device_path,
     if (!pu_emmc_parse_clean(self, root, error))
         return NULL;
 
+    /* Check if raw binary overwrites partitions */
+    if (self->partitions && self->raw) {
+        PuEmmcPartition *part = self->partitions->data;
+        gsize part_start = part->offset * self->device->sector_size;
+        gsize raw_start;
+        gsize raw_end;
+        struct stat st;
+        for (GList *b = self->raw; b != NULL; b = b->next) {
+            PuEmmcBinary *bin = b->data;
+            g_autofree gchar *path = NULL;
+
+            path = pu_path_from_uri(bin->input->uri, prefix, error);
+            if (path == NULL) {
+                g_prefix_error(error, "Failed parsing input URI for partition: ");
+                return NULL;
+            }
+            if (stat(path, &st) == -1) {
+                g_set_error(error, PU_ERROR, PU_ERROR_FAILED,
+                            "Could not get file size of '%s'", bin->input->uri);
+                return NULL;
+            }
+            bin->input->size = st.st_size;
```

Comment:
We should parse the size of raw binaries earlier outside of this statement in `pu_emmc_parse_raw()`, so it is available in other places, too.

Reply:
Done. Also added a new function `pu_emmc_get_input_size()` which can retrieve the size of any `PuEmmcInput`.